app/uk/gov/hmrc/eoricommoncomponent/frontend/services/cache/sessionCache.scala

```diff
@@ -119,17 +120,31 @@ class SessionCache @Inject() (
 
   def saveRegisterWithEoriAndIdResponse(
     rd: RegisterWithEoriAndIdResponse
-  )(implicit request: Request[_]): Future[Boolean] =
-    putData(registerWithEoriAndIdResponseKey, Json.toJson(rd)) map (_ => true)
+  )(implicit request: Request[_]): Future[Boolean] = for {
+    subCompleteDetails <- submissionCompleteDetails
+    _                  <- putData(registerWithEoriAndIdResponseKey, Json.toJson(rd)) map (_ => true)
+    _ <- putData(
+      submissionCompleteKey,
+      Json.toJson(
+        SubmissionCompleteData(subCompleteDetails.subscriptionDetails, Some(rd.responseCommon.processingDate))
+      )
```

Comment:
Should we just be calling saveSubmissionCompleteDetails here?

We can also use `subCompleteDetails.copy(processingDate = rd.responseCommon.processingDate)` here so that if any further data gets added it gets retained. What we want is a copy of the existing object with the processing date updated.

Reply:
Yep, good point - will make these changes to both functions